src/plugins/dashboard/public/application/dashboard_router.tsx

```diff
@@ -229,26 +232,35 @@ export async function mountApp({
         <KibanaContextProvider services={dashboardServices}>
           <presentationUtil.ContextProvider>
             <KibanaThemeProvider theme$={core.theme.theme$}>
-              <HashRouter>
-                <Switch>
-                  <Route
-                    path={[
-                      DashboardConstants.CREATE_NEW_DASHBOARD_URL,
-                      `${DashboardConstants.VIEW_DASHBOARD_URL}/:id`,
-                    ]}
-                    render={renderDashboard}
-                  />
-                  <Route
-                    exact
-                    path={DashboardConstants.LANDING_PAGE_PATH}
-                    render={renderListingPage}
-                  />
-                  <Route exact path="/">
-                    <Redirect to={DashboardConstants.LANDING_PAGE_PATH} />
-                  </Route>
-                  <Route render={renderNoMatch} />
-                </Switch>
-              </HashRouter>
+              <TableListViewKibanaProvider
```

Comment:
Does this provider need to be surrounding the entire dashboard router? Would it be possible to place it only around the listing page?

Reply:
It was a recommendation from @clintandrewhall I think at some point the package provider will be merged in some sort. So I say yes it is the way to go with package providers. I let Clint bring more input here.